### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
Tidy up documentation generation for Hexdocs.pm which leverages on
latest features of ExDoc.

### CODE REVIEW

File: lib/life_cycle_hook.ex [hunk 1]
```diff
@@ -1,4 +1,11 @@
 defmodule LifeCycleHook do
+  @external_resource "README.md"
+  @moduledoc "README.md"
+             |> File.read!()
+             |> String.split("<!-- MDOC !-->")
+             |> Enum.fetch!(1)
```
Comment (nallwhy) [inline lib/life_cycle_hook.ex:6]: Cool!

File: README.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # LifeCycleHook
 
+[![Hex Version](https://img.shields.io/hexpm/v/life_cycle_hook.svg)](https://hex.pm/packages/life_cycle_hook)
+[![Hex Docs](https://img.shields.io/badge/hex-docs-lightgreen.svg)](https://hexdocs.pm/life_cycle_hook/)
+[![Total Download](https://img.shields.io/hexpm/dt/life_cycle_hook.svg)](https://hex.pm/packages/life_cycle_hook)
+[![License](https://img.shields.io/hexpm/l/life_cycle_hook.svg)](https://github.com/nallwhy/life_cycle_hook/blob/master/LICENSE)
```
Comment (nallwhy) [inline README.md]: I think this is the correct path after this commit is merged.

```suggestion
[![License](https://img.shields.io/hexpm/l/life_cycle_hook.svg)](https://github.com/nallwhy/life_cycle_hook/blob/master/LICENSE.md)
```

**Review by nallwhy — COMMENTED** [review]
Thank you for your contribution!

**Comment by nallwhy** [comment]
❤️

**Comment by kianmeng** [comment]
:tada: :tada: :tada: :tada: :tada:

**Pull request merged**